.status.loadBalancer.ingress[0].ip is empty in AWS

ipedrazas opened the pull request:
Fixes #237

`.status.loadBalancer.ingress[0].ip}` has no value in AWS since the ELB only returns the hostname. I've added instructions to get the jenkins url using the hostname of the eLB instead of the IP.

ipedrazas committed:
.status.loadBalancer.ingress[0].ip is empty in AWS

Fixes #237

`.status.loadBalancer.ingress[0].ip}` has no value in AWS since the ELB only returns the hostname. I've added instructions to get the jenkins url using the hostname of the eLB instead of the IP.

ipedrazas commented:
I signed it!

lachie83 commented:
Thanks @ipedrazas. How about we update to this?

```
kubectl describe svc {{ template "fullname" . }} --namespace {{ .Release.Namespace }} | grep Ingress | awk '{print $3}'
```

ipedrazas commented:
Not sure about this.

Have you thought of the windows users? I'm not sure if Windows has `grep` and `awk`, which it might be a problem for some users... but I've just realised that this only makes sense if Helm supports Windows, which is not the case, yet (correct me if I'm wrong).

Another reason would be that my form nudges users to learn the harsh world of kubernetes schemas.

But my point was to provide a copy and paste instruction for users who are not familiar with kubernetes and helm.

lachie83 commented:
All fair points. I would be nice if we could determine what platform the cluster is running on to switch that to a conditional. It's just a bunch of information to go through to get started

prydonius commented:
See https://github.com/kubernetes/charts/issues/84 for other alternatives.

```
kubectl get svc {{ template "fullname" . }} --template "{{ range (index .status.loadBalancer.ingress 0) }}{{.}}{{ end }}"
```
would also work and doesn't require awk/grep.

ipedrazas commented:
Happy to update it so we can have some consistency across charts 😄

lachie83 commented:
Let's go with what @prydonius suggested.

lachie83 commented on stable/jenkins/templates/NOTES.txt:
```diff
@@ -10,8 +10,14 @@
+  
   echo http://$SERVICE_IP:{{ .Values.Master.ServicePort }}/login
+  
+If you're running Kubernetes in AWS, the ingress has not ip, but you can use the Hostname
```
You can yank this block now right?

ipedrazas replied:
yep, I've just realised. Editing this with a phone is not ideal, sorry.

ipedrazas committed:
.status.loadBalancer.ingress[0].ip is empty in AWS

`.status.loadBalancer.ingress[0].ip}` has no value in AWS since the ELB
only returns the hostname. I've added instructions to get the jenkins
url using the hostname of the ELB instead of the IP.

lachie83 commented on stable/jenkins/templates/NOTES.txt:
```diff
@@ -10,8 +10,9 @@
 **** NOTE: It may take a few minutes for the LoadBalancer IP to be available.                      ****
 ****       You can watch the status of by running 'kubectl get svc -w {{ template "fullname" . }}' ****
-  export SERVICE_IP=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ template "fullname" . }} -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+  export SERVICE_IP=$(kubectl get svc {{ template "fullname" . }} --template "{{ range (index .status.loadBalancer.ingress 0) }}{{.}}{{ end }}")
```
Can you please add ` --namespace {{ .Release.Namespace }}` to this?

ipedrazas commented:
Done. Squashed my mess also.

lachie83 commented:
Looks like it's not escaped properly
```
helm install . --namespace=patch
Error: render error in "jenkins/templates/NOTES.txt": template: jenkins/templates/NOTES.txt:13:138: executing "jenkins/templates/NOTES.txt" at <.status.loadBalancer...>: can't evaluate field loadBalancer in type interface {}
```

lachie83 commented:
Here's an example https://github.com/kubernetes/charts/blob/7e971635642494b607118aab56e17a21508bcc7f/stable/ghost/templates/NOTES.txt#L21

ipedrazas committed:
escape fixed

lachie83 commented:
Perfect! Thanks @ipedrazas!

lachie83 commented:
LGTM
